### PR TITLE
PR for Authentication

### PR DESCRIPTION
**Pull request by manojasarsa**
This PR includes basic authentication for users.
It includes:
- SignUp
- Login
- Logout

### CODE REVIEW

**Comment by manojasarsa** [comment]
https://user-images.githubusercontent.com/36642146/160494669-446ca463-0e3f-48ac-9310-d021661b5cee.mp4

File: src/contexts/authContext.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                localStorage.setItem("jwtToken", JSON.stringify({userInfo: createdUser, token: encodedToken}));
+                dispatch({type: "LOGIN", payload: {userInfo: createdUser, token: encodedToken}});
+                // navigate("/");
+            }
+        } catch (error) {
+            
+            // setPwdError(error);
+            // console.log(pwdError);
+            console.log("error recieved");
+        }
```
Comment (gopalbbharadva) [inline src/contexts/authContext.js:43]: Suggestion: can remove comments if not used.

File: src/contexts/authContext.js [hunk 2]
```diff
@@ -1,82 +1,74 @@
...
-
-            const response = await axios.post("/api/auth/signup", {
-                firstName: firstName,
-                lastName: lastName,
-                email: email,
-                password: password,
-            });
-
+            const response = await axios.post("/api/auth/logi", {email, password});
+
```
Comment (gopalbbharadva) [inline src/contexts/authContext.js]: Bug: endpoint has spelling mistake."login*"

**Review by gopalbbharadva — APPROVED** [review]
Other than comments all good and well-structured. 👍

**Review by mimansha-swarup — APPROVED** [review]
code looks clean 👍

File: src/contexts/authContext.js (hunk 2 above)
Comment (manojasarsa) [reply src/contexts/authContext.js]: I think all the new changes are not updated in the code after last commit.

Comment (manojasarsa) [reply src/contexts/authContext.js]: Removed all the comments and bugs as well in the new one.

**Pull request merged**